/*
 *  This is free software: you can redistribute it and/or modify it
 *  under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  any later version.
 *  The software is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *  You should have received a copy of the GNU General Public License
 *  along with the software. If not, see <http://www.gnu.org/licenses/>.
 *
 *  Copyright 2019, Laurens Van Damme.
 */

syntax = "proto3";
package proto;

// Epi-output consists of multiple Locations from a GeoGrid.
message EpiOutput {

    // Location of UUID, name, province and population count, coordinate and it's epi-output per measured day.
    message Location {

        int64 id = 1;
        string name = 2;
        int64 province = 3;
        int64 population = 4;

        // Coordinate with longitude and latitude.
        message Coordinate {
            double longitude = 1;
            double latitude = 2;
        }
        Coordinate coordinate = 5;

        // Age bracket in the epi-output.
        message AgeBracket {
            string type = 1;

            // Health status in the age bracket.
            message HealthStatus {
                string type = 1;

                // Measured percentage of that health category in a certain day.
                message Day {
                    int64 day = 1;
                    double percentage = 2;
                }
                repeated Day days = 2;
            }
            repeated HealthStatus healthStatuses = 2;
        }
        repeated AgeBracket ageBrackets = 6;
    }

    // Measured day in teh epi-output.
//    message Measured_day {
//        int64 day = 1;
//    }

    // The actual message content: multiple Locations and multiple measured days.
    repeated Location locations = 1;
//    repeated Measured_day measured_days = 2;
}
